src/utils/parsers.ts

```diff
@@ -208,6 +208,8 @@ function parseEnvArgs(
     [],
     [],
   ];
+  console.error('val', value)
+  console.error('cur', current)
   if (current[1].length === 0) {
     // Parse a secret path
     if (value !== '--') {
```

Comment:
For now, we are also returning the `'--'` in the returned command arguments as such, but it doesn't seem ideal.

```ts
> // Command setup
> // Calling with `polykey secrets env vault:SECRET -- command`
> console.log(args);
[['vault', 'SECRET', undefined'], ['--', 'command']]
```

Currently, this is being removed at the beginning of the `env` command's action, but this shouldn't be the case in the first place. How should we deal with it, if at all?

One idea I had was to store some state inside `prev` as a JSON object, so we can extract the args. The parser can set and get the previously-set state to switch contexts between parsing vaults to parsing commands.

Another one is to store this state out of the function, so the `prev` doesn't get messed up, but storing the state of a function outside the actual function just feels weird.

Should this be addressed, or the command removing `'--'` is fine and the behaviour doesn't need modification?

Reply:
We can't rely on any checks like `if (current[1].length === 1) { current[1].shift(); }`, as once it has been removed, it will trigger the condition again, and if the user intends to pass `'--'` as an actual value, then this approach will also incorrectly remove that, too. As the array is of strings, I cannot use values like `null` or `undefined` as placeholder values to be removed later, and I also can't use an empty string as the placeholder because `commander` can return any combination of a string.

Reply:
To address this, we are storing a state flag in the parser `prev`. If we encounter the `--`, then we set the flag and continue. This will then result in the following iterations appending to the command list instead. We can simply ignore this state when deconstructing the array anywhere else.